plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -106,9 +108,20 @@ struct amqp_trx_plugin_impl : std::enable_shared_from_this<amqp_trx_plugin_impl>
    }
 
    void on_block_start( uint32_t bn ) {
-      block_uuid = boost::uuids::to_string( boost::uuids::random_generator()() );
-      tracked_block_uuid_rks.clear();
-      trx_queue_ptr->on_block_start();
+      if (!prod_plugin->paused() || allow_speculative_execution) {
+         if (!started_consuming) {
+            ilog("Starting consuming amqp messages during on_block_start");
+            amqp_trx->start_consume(true);
+            started_consuming = true;
+         }
+
+         block_uuid = boost::uuids::to_string( boost::uuids::random_generator()() );
+         tracked_block_uuid_rks.clear();
+         trx_queue_ptr->on_block_start();
+      } else {
+         trx_queue_ptr->clear();
```

Comment:
I'm confused. I thought we were not going to handle the `pause` case for now.
If we are going to handle the `pause` case now then I think we should `reject` the all received amqp messages.
We would also need to stop consuming from the amqp queue.